src/main/java/com/google/research/bleth/simulator/SimulationMetadata.java

```diff
@@ -108,17 +110,27 @@ public static SimulationMetadata read(String simulationId) {
 
     /**
      * Read all existing SimulationMetadata entites from the db, and return them as a hashmap.
+     * @param sortProperty is the name of the property to sort the results by (optional).
+     * @param sortDirection is the direction of the sort (optional).
      * @return a hashmap mapping a simulationId to the corresponding SimulationMetadata object.
      */
-    public static ImmutableMap<String, SimulationMetadata> listSimulations() {
-        ImmutableMap.Builder<String, SimulationMetadata> simulations = new ImmutableMap.Builder<>();
+    public static LinkedHashMap<String, SimulationMetadata> listSimulations(Optional<String> sortProperty,
+                                                                            Optional<Query.SortDirection> sortDirection) {
+        LinkedHashMap<String, SimulationMetadata> simulations = new LinkedHashMap<>();
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Query simulationMetadataQuery = new Query(Schema.SimulationMetadata.entityKind);
+        if (sortProperty.isPresent()) {
+            if (sortDirection.isPresent()) {
```

Comment:
As mentioned in the caller - prefer to just have the caller default `sortDirection` to `ASCENDING`, and then you don't need this extra "if" at all.